### DISCUSSION TITLE
Wikijs - Save document takes a long time - network call takes approx. 6 seconds

### DISCUSSION THREAD

**Discussion by tobidemski** [requarks/wiki#4982] [Error / Bug Report]
Hey guys,

Im using wikijs ~~2.5.274~~ 2.5.297 and having a few pages (not much content) (< 20) and I'm the only user.
After adding / saving a page it takes a long time till the process is finished.

Network tab after clicking the save button 3 times: Each save takes around 6second to finish the save call.
![image](https://user-images.githubusercontent.com/17470007/152891246-1eef9b70-d131-43fd-b48d-a28ec5851687.png)

Hardware Intel NUC:
- Intel® Core™ i7-10710U Prozessor (bis zu 4,70 GHz), Hexa-Core
- WD Blue SN550 NVMe SSD 2 TB PCIe M.2 2280
- 64GB RAM
- 
Running wikijs in docker through vm managed by Proxmox. (Addressed 2 cores and 4gb ram to this instance)

Anyone had the same problem and now whats causing the issue?
Would be nice to get the speed up for my instance!

Best regards

**Comment by crztssr** [comment]
Hey. Faced same behavior, however I am running standalone (not dockerized) wiki.js along with postgresql in cloud 2CPU/4GB RAM vm with SSD.
I've looked into, and found that while saving the page there is 1CPU is used on 100% for dozen of seconds. 
I have GIT setup as a backup solution, and the same behavior is observed when I press force sync.
As far as I understood wiki.js tries to read through all of it's files adding it to local repository before syncing to github.
But for some reason the issue is intermittent and sometimes saving and syncing tooks couple of seconds, and sometimes it takes more than 20seconds creating a pretty high load on the server. 
Restarting wiki.js service won't help.

Will try to get some more debug details.

@tobidemski do you also have git as backup solution?

**Reply by tobidemski** [reply]
Hey @crztssr,
no I don't use any backup solution so far because of this slowness of the system. As long as this is such slow I don't use wikijs that often because it makes no fun to edit pages.

**Comment by fumbles** [comment]
How many pages do you have?

We had this issue as well-have 60,000 pages.... had to use a patch which I made [PR#6082 ](https://github.com/requarks/wiki/pull/6082) which includes it

**Comment by fumbles** [comment]
[patches.zip](https://github.com/requarks/wiki/files/11009053/patches.zip)
![image](https://user-images.githubusercontent.com/90968/226120559-3109cd15-8349-47bd-919a-6204f8e7d73b.png)
